### DISCUSSION TITLE
Caddy Config for WEB_DOMAIN Variable?

### DISCUSSION THREAD

**Discussion by pellux-network** [mastodon/mastodon#20769] [Q&A]
I'm using Caddy as my reverse proxy. This is my current config:

```
https://domain.com {
	@local {
		file
		not path /
	}
	@local_media {
		path_regexp /system/(.)
	}
	@streaming {
		path /api/v1/streaming/
	}
	@cache_control {
		path_regexp ^/(emoji|packs|/system/accounts/avatars|/system/media_attachments/files)
	}

	root * /home/mastodon/live/public
	log {
		output file /var/log/caddy/mastodon.log
	}

	encode zstd gzip

	handle_errors {
		rewrite 500.html
		file_server
	}

	header {
		Strict-Transport-Security "max-age=31536000"
	}
	header /sw.js Cache-Control "public, max-age=0"
	header @cache_control Cache-Control "public, max-age=31536000, immutable"

	handle @local {
		file_server
	}

	reverse_proxy @streaming {
		to http://192.168.86.38:4000

		transport http {
			keepalive 5s
			keepalive_idle_conns 10
		}
	}

	reverse_proxy {
		to http://192.168.86.38:3000

		header_up X-Forwarded-Port 443
		header_up X-Forwarded-Proto https

		transport http {
			keepalive 5s
			keepalive_idle_conns 10
		}
	}
}
```
I'm hosting Mastodon on a .social domain but I want the users to be on the .net domain however I'm not quite sure how to go about this. Does anyone know what I need to add to my config to accomplish this?

**Comment by madeofstown** [comment]
As far as I know, you will need to forward the path "/.well-known/webfinger" on your domain.net to redirect to "https://domain.social$URI". I've never used caddy though so you'll need to look up how to do a 301 redirect with it.

**Reply by pellux-network** [reply]
I looked up redirects for Caddy and this is the syntax
`redir [<matcher>] <to> [<code>]`

I know you said you weren't familiar with Caddy but given that and the rest of my config, does this seem right to you?

```
domain.net/.well-known/webfinger {
	redir https://domain.social$request_uri 301
}
```

Unfortunately, I'm also not sure with $request_uri can be left as is as you have just $URI

**Reply by madeofstown** [reply]
I simply put $URI because I wasn't sure of the variable that Caddy uses. It looks like your redirect should work based on what you've quoted for Caddy redirect instructions.

**Comment by ineffyble** [comment]
There's some work to try and provide a canonical Caddyfile to help admins in future: https://github.com/mastodon/mastodon/pull/20645

**Reply by pellux-network** [reply]
This is definitely a step in the right direction. Although, after further digging it turns out I can't even access the instance on localhost, I just get a 502 error. I'm assuming the real problem here is I have no clue how the mastodon nginx is supposed to be configured with a reverse proxy. I have an Ubuntu VM within Windows Server running Mastodon, the Caddy server which is acting as the reverse proxy server for all of services is on the main Windows install, the Windows and Ubuntu installs have their own IPs on the LAN. How should I config the mastodon nginx server on Ubuntu, so that it will work with the reverse proxy?

**Reply by madeofstown** [reply]
Because your VM has its own local IP you would be better off skipping caddy altogether and just using Nginx inside the VM. If you can't (public IP points to windows server) then you should redirect all traffic for "mastodon.domain" that arrives at the windows server to the local IP of your Ubuntu VM using whichever reverse proxy you choose. If you don't know how to do a 301 redirect using Caddy then use Nginx because instructions for how to do redirecting with Nginx are readily available.
